### ISSUE TITLE
Builds for OSX Pyhton 3.9 and 3.10 builds failing

### ISSUE THREAD

**Issue by mauzey1** [conda-forge/cmor-feedstock#47]
There is currently an issue with the CMOR OSX Python 3.9 and 3.10 builds where it is running into the following error while building.

```
Installed $PREFIX/lib/python3.10/site-packages/CMOR-3.6.1-py3.10-macosx-10.9-x86_64.egg
Processing dependencies for CMOR==3.6.1
Searching for CMOR==3.6.1
Reading https://pypi.org/simple/CMOR/
Couldn't find index page for 'CMOR' (maybe misspelled?)
Scanning index of all packages (this may take a while)
Reading https://pypi.org/simple/
No local packages or working download links found for CMOR==3.6.1
error: Could not find suitable distribution for Requirement.parse('CMOR==3.6.1')
make: *** [Makefile:150: python] Error 1
Traceback (most recent call last):
  File "/Users/runner/miniforge3/bin/conda-mambabuild", line 10, in <module>
    sys.exit(main())
  File "/Users/runner/miniforge3/lib/python3.9/site-packages/boa/cli/mambabuild.py", line 239, in main
    call_conda_build(action, config)
  File "/Users/runner/miniforge3/lib/python3.9/site-packages/boa/cli/mambabuild.py", line 211, in call_conda_build
    result = api.build(
  File "/Users/runner/miniforge3/lib/python3.9/site-packages/conda_build/api.py", line 186, in build
    return build_tree(
  File "/Users/runner/miniforge3/lib/python3.9/site-packages/conda_build/build.py", line 3088, in build_tree
    packages_from_this = build(metadata, stats,
  File "/Users/runner/miniforge3/lib/python3.9/site-packages/conda_build/build.py", line 2211, in build
    utils.check_call_env(cmd, env=env, rewrite_stdout_env=rewrite_env,
  File "/Users/runner/miniforge3/lib/python3.9/site-packages/conda_build/utils.py", line 411, in check_call_env
    return _func_defaulting_env_to_os_environ('call', *popenargs, **kwargs)
  File "/Users/runner/miniforge3/lib/python3.9/site-packages/conda_build/utils.py", line 391, in _func_defaulting_env_to_os_environ
    raise subprocess.CalledProcessError(proc.returncode, _args)
subprocess.CalledProcessError: Command '['/bin/bash', '-o', 'errexit', '/Users/runner/miniforge3/conda-bld/cmor_1659725740843/work/conda_build.sh']' returned non-zero exit status 2.
##[error]Bash exited with code '1'.
Finishing: Run OSX build
```

I didn't see anything while looking through all of the preceding build steps where an error could have occurred.  This issue was also happening to CMOR's nightly builds since they use the same recipe as this feedstock.  A diff of the conda build output from CircleCI during the byte-compiling and copying part before the error occurs might provide a clue.

What the last successful build contains:
```
...
byte-compiling build/bdist.macosx-10.9-x86_64/egg/cmor/Test/jamie_positive.py to jamie_positive.cpython-310.pyc
byte-compiling build/bdist.macosx-10.9-x86_64/egg/cmor/Test/test_python_CMIP6_wrong_activity.py to test_python_CMIP6_wrong_activity.cpython-310.pyc
byte-compiling build/bdist.macosx-10.9-x86_64/egg/cmor/Test/test_python_reverted_lats.py to test_python_reverted_lats.cpython-310.pyc
/Users/distiller/project/workdir/macos/miniconda/conda-bld/cmor_1659395216019/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/lib/python3.10/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(
/Users/distiller/project/workdir/macos/miniconda/conda-bld/cmor_1659395216019/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/lib/python3.10/site-packages/setuptools/command/easy_install.py:144: EasyInstallDeprecationWarning: easy_install command is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(
byte-compiling build/bdist.macosx-10.9-x86_64/egg/cmor/Test/test_python_joerg_9.py to test_python_joerg_9.cpython-310.pyc
byte-compiling build/bdist.macosx-10.9-x86_64/egg/cmor/Test/test_python_CMIP6_CV_badsourcetypeRequired.py to test_python_CMIP6_CV_badsourcetypeRequired.cpython-310.pyc
...
```

Same section on a recent failed build:
```
...
byte-compiling build/bdist.macosx-10.9-x86_64/egg/cmor/Test/jamie_positive.py to jamie_positive.cpython-310.pyc
byte-compiling build/bdist.macosx-10.9-x86_64/egg/cmor/Test/test_python_CMIP6_wrong_activity.py to test_python_CMIP6_wrong_activity.cpython-310.pyc
byte-compiling build/bdist.macosx-10.9-x86_64/egg/cmor/Test/test_python_reverted_lats.py to test_python_reverted_lats.cpython-310.pyc
error: Could not find suitable distribution for Requirement.parse('CMOR==3.6.1')
make: *** [Makefile:150: python] Error 1
byte-compiling build/bdist.macosx-10.9-x86_64/egg/cmor/Test/test_python_joerg_9.py to test_python_joerg_9.cpython-310.pyc
byte-compiling build/bdist.macosx-10.9-x86_64/egg/cmor/Test/test_python_CMIP6_CV_badsourcetypeRequired.py to test_python_CMIP6_CV_badsourcetypeRequired.cpython-310.pyc
...
```
Is there some deprecated setup.py or easy_install command in this feedstock that is no longer supported?

**Comment by mauzey1** [comment]
@xylar @ocefpaf Do you have any insight into what could be going wrong?

Also, I ended up accidentally creating new conda-forge builds from a test branch I pushed to this repo.  They were created from Linux builds and two other OSX builds that didn't fail.  Should those be deleted?

**Comment by xylar** [comment]
@mauzey1, regarding making a test branch, it isn't safe to push branches to the main repo because it does, indeed create packages.  These should be marked as broken.  Ask for help from someone form conda-forge core for that or look at the documentation.

I don't have any insight into the build problems you're having, I'm afraid.  This package doesn't get built in the same way as typical python packages (it has a `./configure`, whereas most python packages use `setup.py`, so I think you're kind of on your own with this pretty unconventional approach).

**Comment by xylar** [comment]
@mauzey1, sorry, I got pretty frustrated with how different `cmor` is than other packages.

I believe the problem is that you are not installing `cmor` without checking for missing dependendencies:
https://github.com/PCMDI/cmor/blob/0748f2c1d3b1539313691d05cd4c279184cd7db1/Makefile.in#L150
If this were:
```
	LDSHARED=${LDSHARED} ${PYTHONEXEC} setup.py install  --no-deps @CDATPREFIX@ 
```

**Comment by mauzey1** [comment]
This issue has been solved in CMOR 3.7.0, which was released today.
